### PR TITLE
Add default limit to OLED dirty processing

### PR DESCRIPTION
**Pull request by zvecr**
## Description

In response to <https://github.com/qmk/qmk_firmware/pull/18887#issuecomment-1312812389>.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by sigprof — APPROVED** [review]
Looks reasonable; works on STM32F401 (did not actually test the “always dirty” case though, which is what prompted the original PR).

File: drivers/oled/ssd1306_sh1106.c [hunk 1]
```diff
@@ -301,7 +301,8 @@ void oled_render(void) {
     oled_on();
 
     uint8_t update_start = 0;
-    while (oled_dirty) { // render all dirty blocks
+    uint8_t dirty_count  = 0;
+    while (oled_dirty && dirty_count++ < OLED_UPDATE_PROCESS_LIMIT) { // render all dirty blocks (up to the configured limit)
```
Comment (fauxpark) [inline drivers/oled/ssd1306_sh1106.c:305]: Is `oled_dirty` still needed with this new variable? I would think the same could be expressed by `dirty_count` being nonzero.

Comment (zvecr) [reply drivers/oled/ssd1306_sh1106.c]: We still need to track 2 things,
- which blocks are dirty
- how many have been processed in a given call to `oled_render`

Ive renamed the variable to maybe convey that slightly better.

Comment (fauxpark) [reply drivers/oled/ssd1306_sh1106.c]: Ah I see, I thought `oled_dirty` was a bool.

**Pull request merged**